src/Http/Http.Abstractions/src/RouteHandlerInvocationContext.cs

```diff
@@ -23,13 +30,24 @@ public RouteHandlerInvocationContext(HttpContext httpContext, params object[] pa
     /// <summary>
     /// The <see cref="HttpContext"/> associated with the current request being processed by the filter.
     /// </summary>
-    public HttpContext HttpContext { get; }
+    public virtual HttpContext HttpContext { get; }
 
     /// <summary>
     /// A list of parameters provided in the current request to the filter.
     /// <remarks>
     /// This list is not read-only to permit modifying of existing parameters by filters.
     /// </remarks>
     /// </summary>
-    public IList<object?> Parameters { get; }
+    public virtual IList<object?> Parameters { get; }
+
+    /// <summary>
+    /// Retrieve the parameter given its position in the argument list.
+    /// </summary>
+    /// <typeparam name="T">The <see cref="Type"/> of the resolved parameter.</typeparam>
+    /// <param name="index">An integer representing the position of the parameter in the argument list.</param>
+    /// <returns>The parameter at a given <paramref name="index"/></returns>
+    public virtual T GetParameter<T>(int index)
+    {
+        return (T)Parameters[index]!;
```

Comment:
What happens if given a handler like `(string? one) => ...`, and `one` is null, but a filter calls `context.GetArgument<string>(0)`?

Do they get `null` even though we declared the return type a `T` and not `T?`? Change the declared return type to `T?`. Or do we manually throw an `InvalidCastException`?

Of course, the caller should call `context.GetArgument<string?>(0)` instead of `context.GetArgument<string>(0)` in this case. But I'm just thinking about what happens when people mess this up. If the parameter was a value type like `int?` instead of `string?`, I think it'd already throw `InvalidCastException`.

Reply:
We discussed the challenge of adding nullability annotations on a generic type in an API that was unconstrained and could potentially return value or reference types. For this particular scenario, the API will actually return an NRE if the cast is invalid. However, there is a possibility that we could use an analyzer in the future to warn about this behavior in particular scenarios. Opened https://github.com/dotnet/aspnetcore/issues/41489 to track.